docs/pipelines.md

```diff
@@ -316,8 +317,60 @@ tasks:
       name: build-push
 ```
 
+### Guard `Task` execution using `WhenExpressions`
```

Comment:
we need more doc on the order of tasks execution (from the TEP) i.e. `runAfter` is not necessary when an `input` is a task result otherwise `runAfter` is must and no task ordering guaranteed .

Reply:
nope, sorry if I am causing any confusion, what I meant to add here was "When expression with task results introduces resource dependency where as when expression with params does not change task execution order. A pipeline task with `runAfter` and `When` expression with task results both play role in execution order." does it read well? 🤔

Reply:
yes yes, that makes sense...thanks for the clarification!